lib/features/recurring_donations/create/cubit/create_recurring_donation_cubit.dart

```diff
@@ -0,0 +1,250 @@
+import 'dart:io';
+
+import 'package:bloc/bloc.dart';
+import 'package:equatable/equatable.dart';
+import 'package:givt_app/core/failures/failure.dart';
+import 'package:givt_app/core/logging/logging.dart';
+import 'package:givt_app/features/recurring_donations/create/models/recurring_donation.dart';
+import 'package:givt_app/features/recurring_donations/create/models/recurring_donation_frequency.dart';
+import 'package:givt_app/features/recurring_donations/create/repositories/create_recurring_donation_repository.dart';
+import 'package:givt_app/shared/models/collect_group.dart';
+
+part 'create_recurring_donation_state.dart';
+
+class CreateRecurringDonationCubit extends Cubit<CreateRecurringDonationState> {
+  CreateRecurringDonationCubit(
+    this._createRecurringDonationRepository,
+  ) : super(
+          CreateRecurringDonationState(
+            startDate: DateTime.now().add(
+              const Duration(days: 1),
+            ),
+            endDate: DateTime.now().add(
+              const Duration(days: 2),
+            ),
+          ),
+        );
+
+  final CreateRecurringDonationRepository _createRecurringDonationRepository;
+
+  void setRecipient(CollectGroup recipient) {
+    emit(
+      state.copyWith(
+        status: CreateRecurringDonationStatus.fieldChanged,
+        recipient: recipient,
+      ),
+    );
+  }
+
+  void setAmount(double amount) {
```

Comment:
shall we check some max value here? the current flow does not work with such big amounts like 33333.0